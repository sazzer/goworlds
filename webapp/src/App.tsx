import React from 'react';
import { Container } from 'semantic-ui-react';

function App() {
  return (
      <Container>
        Hello, World
      </Container>
  );
}

export default App;
