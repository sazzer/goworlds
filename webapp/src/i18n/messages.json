{
  "page": {
    "header": "Application Name"
  },
  "loginArea": {
    "header": "Login/Register",
    "email": {
      "label": "Email Address",
      "placeholder": "Enter Email Address",
      "errors": {
        "required": "Please enter an Email Address",
        "email": "Please enter a valid Email Address"
      }
    },
    "name": {
      "label": "Name",
      "placeholder": "Enter Name",
      "errors": {
        "required": "Please enter a user name"
      }
    },
    "password": {
      "label": "Password",
      "placeholder": "Enter Password",
      "errors": {
        "required": "Please enter a password"
      }
    },
    "password2": {
      "label": "Re-enter Password",
      "placeholder": "Enter Password",
      "errors": {
        "required": "Please re-enter the password",
        "match": "Passwords do not match"
      }
    },
    "submit": {
      "loginRegister": "Login/Register",
      "login": "Login",
      "register": "Register",
      "cancel": "Cancel",
      "errors": {
        "unexpected_error": "An unexpected error occurred",
        "access_denied": "Invalid password or account blocked",
        "tag:goworlds,2019:users/problems/duplicate-email-address": "Email Address is already registered"
      }
    }
  },
  "userMenu": {
    "editProfile": "Edit Profile",
    "logOut": "Log Out"
  },
  "profile": {
    "loading": "Loading...",
    "breadcrumbs": {
      "home": "Home",
      "profile": "Profile: <1>{{name}}</1>"
    },
    "menu": {
      "profile": "Profile",
      "password": "Change Password"
    },
    "form": {
      "email": {
        "label": "Email Address",
        "placeholder": "Enter Email Address",
        "errors": {
          "required": "Please enter an Email Address",
          "email": "Please enter a valid Email Address"
        }
      },
      "name": {
        "label": "Name",
        "placeholder": "Enter Name",
        "errors": {
          "required": "Please enter a user name"
        }
      },
      "submit": "Save Changes",
      "reset": "Reset Changes",
      "errors": {

      }
    }
  }
}
