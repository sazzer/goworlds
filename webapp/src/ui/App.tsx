import React from 'react';
import {Container} from "semantic-ui-react";
import { useTranslation } from 'react-i18next';
import {Route, Link} from 'react-router-dom';

/**
 * The main UI structure
 * @constructor
 */
export default function App () {
    const { t } = useTranslation();

    return (
        <Container>
            <Route path="/" exact render={props => (
                <Link to="/hello">Hi there</Link>
            )} />
            <Route path="/hello" exact render={props => (
                <div>{t('hello')}</div>
            )} />
        </Container>
    );
}
