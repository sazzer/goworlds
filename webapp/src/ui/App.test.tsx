import '../i18n';

import React from 'react';
import {shallow} from 'enzyme';
import App from './App';


/** Set up the component to test */
function setup() {
    return {
        element: shallow(<App />)
    };
}

it('renders without crashing', () => {
    const {element} = setup();

    expect(element).toMatchSnapshot();
});
