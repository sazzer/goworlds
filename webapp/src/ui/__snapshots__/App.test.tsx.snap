// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Container>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/hello"
    render={[Function]}
  />
</Container>
`;
